### DISCUSSION TITLE
How to set a header on the response in the event of an error?

### DISCUSSION THREAD

**Discussion by steve-marmalade** [mirumee/ariadne#1127] [Q&A]
Hi team, my use-case is that if there is an error during the processing of the GraphQL request, I would like to set the `Cache-Control` header to `no-store` so that the response is not cached.

I see in the Starlette docs that I can add headers to the response like [in this example](https://www.starlette.io/middleware/#inspecting-or-modifying-the-response), but the use-case I described above is more complicated because I would need to parse the response body (which might be streamed in chunks?) and check if there is an "errors" key.

I thought the [Extension.has_errors](https://ariadnegraphql.org/docs/next/types-reference#has_errors) lifecycle hook would be perfect for this, but the context object that is provided does not have `response` object, so I am not sure how to set the header.

Any advice would be appreciated! :pray:

**Comment by rafalp** [comment]
Hi.

I would recommend creating custom HTTP handler for ASGI app that customizes the [`create_json_response`](https://github.com/mirumee/ariadne/blob/master/ariadne/asgi/handlers/http.py#L339C15-L339C35) logic with behaviour you are looking for.
